### PR TITLE
WF-65244-UG documentation for Markdown to Word Conversion

### PR DESCRIPTION
**Pull request by sharonaumao**
<table>
<tr>
<td>
Task Link
</td>
<td>
<a href="https://syncfusion.atlassian.net/browse/WF-65244">https://syncfusion.atlassian.net/browse/WF-65244</a>
</td>
</tr>
<tr>
<td>
Description
</td>
<td>
Add UG document for Markdown to Word conversion
</td>
</tr>
<td>
Content review confirmation
</td>
<td>
 Done
</td>
</tr>
</table>

### CODE REVIEW

File: File-Formats/DocIO/convert-markdown-to-word-document-in-csharp.md [hunk 1]
```diff
@@ -0,0 +1,576 @@
...
+{% endhighlight %}
+
+{% highlight c# tabtitle="UWP" %}
+private static void MdImportSettings_ImageNodeVisited(object sender, Syncfusion.Office.Markdown.MdImageNodeVisitedEventArgs args)
+{
+    //Set image stream based on image name from input Markdown
+    if (args.Uri == "Image_1.png")
+    {
+        Assembly assembly = typeof(App).GetTypeInfo().Assembly;
+        args.ImageStream = assembly.GetManifestResourceStream("Sample.Assets.Image_1.png");
```
Comment (MohanaselvamJothi) [inline File-Formats/DocIO/convert-markdown-to-word-document-in-csharp.md:233]: Use the sample name 
"Customize_image.Assets.Image_1.png"

Address the same feedbacks for uwp, xamarin

**Review by MohanaselvamJothi — CHANGES_REQUESTED** [review]

File: File-Formats/DocIO/convert-markdown-to-word-document-in-csharp.md [hunk 2]
```diff
@@ -0,0 +1,575 @@
+---
+title: Convert Markdown to Word document in C# | DocIO | Syncfusion
```
Comment (Chrispineagunja) [inline File-Formats/DocIO/convert-markdown-to-word-document-in-csharp.md:2]: Title: Convert Markdown to Word document in C# | DocIO | Syncfusion

File: File-Formats/DocIO/convert-markdown-to-word-document-in-csharp.md [hunk 3]
```diff
@@ -0,0 +1,575 @@
...
+description: Convert Markdown to Word document in C# using Syncfusion .NET Word (DocIO) library without Microsoft Word or interop dependencies
```
Comment (Chrispineagunja) [inline File-Formats/DocIO/convert-markdown-to-word-document-in-csharp.md:3]: Description: Convert Markdown to Word document in C# using the Syncfusion .NET Word (DocIO) library without Microsoft Word or interop dependencies.

File: File-Formats/DocIO/convert-markdown-to-word-document-in-csharp.md [hunk 4]
```diff
@@ -0,0 +1,575 @@
...
+platform: file-formats
```
Comment (Chrispineagunja) [inline File-Formats/DocIO/convert-markdown-to-word-document-in-csharp.md:4]: Platform: File-formats

File: File-Formats/DocIO/convert-markdown-to-word-document-in-csharp.md [hunk 5]
```diff
@@ -0,0 +1,575 @@
...
+control: DocIO
```
Comment (Chrispineagunja) [inline File-Formats/DocIO/convert-markdown-to-word-document-in-csharp.md:5]: Control: DocIO

File: File-Formats/DocIO/convert-markdown-to-word-document-in-csharp.md [hunk 6]
```diff
@@ -0,0 +1,575 @@
...
+documentation: UG
```
Comment (Chrispineagunja) [inline File-Formats/DocIO/convert-markdown-to-word-document-in-csharp.md:6]: Documentation: UG

File: File-Formats/DocIO/convert-markdown-to-word-document-in-csharp.md [hunk 7]
```diff
@@ -0,0 +1,575 @@
...
+---
+
+# Markdown to Word Conversion
+
+Markdown is a lightweight markup language that adds formatting elements to plain text documents. The .NET Word (DocIO) library supports to convert Markdown to Word document and vice versa, which mostly follows the CommonMark specification and GitHub-flavored syntax.
```
Comment (Chrispineagunja) [inline File-Formats/DocIO/convert-markdown-to-word-document-in-csharp.md:11]: Markdown is a lightweight markup language that adds formatting elements to plain text documents. The .NET Word (DocIO) library supports converting Markdown to Word document and vice versa, which mostly follows the CommonMark specification and GitHub-flavored syntax.

File: File-Formats/DocIO/convert-markdown-to-word-document-in-csharp.md [hunk 8]
```diff
@@ -0,0 +1,575 @@
...
+
+## Convert Markdown to Word document
+
+Convert an existing markdown file to Word document (DOC, DOCX and RTF) using the .NET Word (DocIO) library.
```
Comment (Chrispineagunja) [inline File-Formats/DocIO/convert-markdown-to-word-document-in-csharp.md:15]: Convert an existing markdown file to a Word document (DOC, DOCX, and RTF) using the .NET Word (DocIO) library.

File: File-Formats/DocIO/convert-markdown-to-word-document-in-csharp.md [hunk 9]
```diff
@@ -0,0 +1,575 @@
...
+    //Load the file stream into a Markdown file.
+    using (WordDocument document = new WordDocument(docStream, FormatType.Markdown))
+    {
+        //Save as a Word document into the MemoryStream.
+        MemoryStream stream = new MemoryStream();
+        await document.SaveAsync(stream, FormatType.Docx);
+        //Save the stream as a Word document in the local machine.
+        Save(stream, "MarkdownToWord.docx");
+    }
+//Please refer to the following link to save the Word document in UWP platform
```
Comment (Chrispineagunja) [inline File-Formats/DocIO/convert-markdown-to-word-document-in-csharp.md:50]: //Please refer to the following link to save a Word document in the UWP platform.

File: File-Formats/DocIO/convert-markdown-to-word-document-in-csharp.md [hunk 10]
```diff
@@ -0,0 +1,575 @@
...
+using (FileStream docStream = new FileStream("Input.md", FileMode.Open, FileAccess.Read))
+{
+    //Load the file stream into a Markdown file.
+    using (WordDocument document = new WordDocument(docStream, FormatType.Markdown))
+    {
+        //Save as a Word document into the MemoryStream.
+        MemoryStream outputStream = new MemoryStream();
+        document.Save(outputStream, FormatType.Docx);
+        outputStream.Position = 0;
+        //Download the Word document in the browser.
```
Comment (Chrispineagunja) [inline File-Formats/DocIO/convert-markdown-to-word-document-in-csharp.md:66]: //Download a Word document in the browser.

File: File-Formats/DocIO/convert-markdown-to-word-document-in-csharp.md [hunk 11]
```diff
@@ -0,0 +1,575 @@
...
+        //Save the stream as a file in the device and invoke it for viewing. 
+        Xamarin.Forms.DependencyService.Get<ISave>().SaveAndView("MarkdownToWord.docx", "application/msword", outputStream);
+    }
+//Please download the helper files from the following link to save the stream as a file and open the file for viewing in the Xamarin platform.
+//https://help.syncfusion.com/file-formats/docio/create-word-document-in-xamarin#helper-files-for-xamarin
+}
+{% endhighlight %}
+{% endtabs %}
+
+T> You can also save the markdown file as [HTML](https://help.syncfusion.com/file-formats/docio/html), [PDF](https://help.syncfusion.com/file-formats/docio/word-to-pdf) and [Image](https://help.syncfusion.com/file-formats/docio/word-to-image).
```
Comment (Chrispineagunja) [inline File-Formats/DocIO/convert-markdown-to-word-document-in-csharp.md:91]: T> You can also save the markdown file as [HTML](https://help.syncfusion.com/file-formats/docio/html), [PDF](https://help.syncfusion.com/file-formats/docio/word-to-pdf), and [Image](https://help.syncfusion.com/file-formats/docio/word-to-image).

File: File-Formats/DocIO/convert-markdown-to-word-document-in-csharp.md [hunk 12]
```diff
@@ -0,0 +1,575 @@
...
+
+## Customize image data
+
+The .NET Word (DocIO) library provides an MdImageNodeVisited event, which is used to customize image data while importing Markdown file. You can implement the logic to customize the image data by using this MdImageNodeVisited event.
```
Comment (Chrispineagunja) [inline File-Formats/DocIO/convert-markdown-to-word-document-in-csharp.md:95]: The .NET Word (DocIO) library provides a MdImageNodeVisited event, which customizes image data while importing a Markdown file. Implement the logic to customize the image data by using this MdImageNodeVisited event.

File: File-Formats/DocIO/convert-markdown-to-word-document-in-csharp.md [hunk 13]
```diff
@@ -0,0 +1,575 @@
...
+
+The following code example shows how to load image data based on image source path when importing the Markdown files.
```
Comment (Chrispineagunja) [inline File-Formats/DocIO/convert-markdown-to-word-document-in-csharp.md:97]: The following code example shows how to load the image data based on the image source path when importing the Markdown files.

File: File-Formats/DocIO/convert-markdown-to-word-document-in-csharp.md [hunk 14]
```diff
@@ -0,0 +1,575 @@
...
+{
+    //Hook the event to customize the image while importing Markdown.
+    document.MdImportSettings.ImageNodeVisited += MdImportSettings_ImageNodeVisited;
+    //Open the Markdown file.
+    document.Open("Input.md");
+
+    //Save as a Word document to the MemoryStream.
+    MemoryStream stream = new MemoryStream();
+    await document.SaveAsync(stream, FormatType.Docx);
+    //Save the stream as a Word document in local machine.
```
Comment (Chrispineagunja) [inline File-Formats/DocIO/convert-markdown-to-word-document-in-csharp.md:137]: //Save the stream as a Word document in the local machine.

File: File-Formats/DocIO/convert-markdown-to-word-document-in-csharp.md [hunk 15]
```diff
@@ -0,0 +1,575 @@
...
+    Save(stream, "Sample.docx");
+    //Please refer to the following link to save the Word document in UWP platform
```
Comment (Chrispineagunja) [inline File-Formats/DocIO/convert-markdown-to-word-document-in-csharp.md:139]: //Please refer to the following link to save a Word document in the UWP platform.

File: File-Formats/DocIO/convert-markdown-to-word-document-in-csharp.md [hunk 16]
```diff
@@ -0,0 +1,575 @@
...
+    document.MdImportSettings.ImageNodeVisited += MdImportSettings_ImageNodeVisited;
+    //Open the Markdown file.
+    document.Open("Input.md");
+
+    //Save as a Word document to the MemoryStream.
+    MemoryStream outputStream = new MemoryStream();
+    document.Save(outputStream, FormatType.Docx);
+    //Save the stream as a file in the device and invoke it for viewing. 
+    Xamarin.Forms.DependencyService.Get<ISave>().SaveAndView("Sample.docx", "application/msword", outputStream);
+    //Please download the helper files from the following link to save the stream as a file and open the file for viewing in Xamarin platform
```
Comment (Chrispineagunja) [inline File-Formats/DocIO/convert-markdown-to-word-document-in-csharp.md:176]: //Please download the helper files from the following link to save the stream as a file and open the file for viewing in the Xamarin platform.

File: File-Formats/DocIO/convert-markdown-to-word-document-in-csharp.md [hunk 17]
```diff
@@ -0,0 +1,575 @@
...
+    //https://help.syncfusion.com/file-formats/docio/create-word-document-in-xamarin#helper-files-for-xamarin
+}
+{% endhighlight %}
+{% endtabs %}
+
+The following code examples show the event handler to customize the image based on source path.
```
Comment (Chrispineagunja) [inline File-Formats/DocIO/convert-markdown-to-word-document-in-csharp.md:182]: The following code examples show the event handler to customize the image based on the source path.

File: File-Formats/DocIO/convert-markdown-to-word-document-in-csharp.md [hunk 18]
```diff
@@ -0,0 +1,575 @@
...
+{% endhighlight %}
+
+{% highlight vb.net tabtitle="VB.NET" %}
+Private Shared Sub MdImportSettings_ImageNodeVisited(ByVal sender As Object, ByVal args As Syncfusion.Office.Markdown.MdImageNodeVisitedEventArgs)
+    'Set the image stream based on the image name from the input Markdown.
+    If args.Uri Is "Image_1.png" Then
+        args.ImageStream = New FileStream("Image_1.png", FileMode.Open)
+    ElseIf args.Uri Is "Image_2.png" Then
+        args.ImageStream = New FileStream("Image_2.png", FileMode.Open)
+    'Retrive the image from website and use it.
```
Comment (Chrispineagunja) [inline File-Formats/DocIO/convert-markdown-to-word-document-in-csharp.md:213]: 'Retrieve the image from the website and use it.

File: File-Formats/DocIO/convert-markdown-to-word-document-in-csharp.md [hunk 19]
```diff
@@ -0,0 +1,575 @@
...
+    {
+        Assembly assembly = typeof(App).GetTypeInfo().Assembly;
+        args.ImageStream = assembly.GetManifestResourceStream("Customize_image.Assets.Image_1.png");
+    }
+    else if (args.Uri == "Image_2.png") 
+    {
+        Assembly assembly = typeof(App).GetTypeInfo().Assembly;
+        args.ImageStream = assembly.GetManifestResourceStream("Customize_image.Assets.Image_2.png");
+    }
+    //Retrive the image from the website and use it.
```
Comment (Chrispineagunja) [inline File-Formats/DocIO/convert-markdown-to-word-document-in-csharp.md:239]: //Retrieve the image from the website and use it.

File: File-Formats/DocIO/convert-markdown-to-word-document-in-csharp.md [hunk 20]
```diff
@@ -0,0 +1,575 @@
...
+
+{% highlight c# tabtitle="ASP.NET Core" %}
+private static void MdImportSettings_ImageNodeVisited(object sender, Syncfusion.Office.Markdown.MdImageNodeVisitedEventArgs args)
+{
+    //Set the image stream based on the image name from the input Markdown.
+    if (args.Uri == "Image_1.png")
+        args.ImageStream = new FileStream("Image_1.png", FileMode.Open);
+    else if (args.Uri == "Image_2.png")
+        args.ImageStream = new FileStream("Image_2.png", FileMode.Open);
+    //Retrive the image from the website and use it.
```
Comment (Chrispineagunja) [inline File-Formats/DocIO/convert-markdown-to-word-document-in-csharp.md:260]: //Retrieve the image from the website and use it.

File: File-Formats/DocIO/convert-markdown-to-word-document-in-csharp.md [hunk 21]
```diff
@@ -0,0 +1,575 @@
...
+    {
+        Assembly assembly = typeof(App).GetTypeInfo().Assembly;
+        args.ImageStream = assembly.GetManifestResourceStream("Customize_image.Assets.Image_1.png");
+    }
+    else if (args.Uri == "Image_2.png") 
+    {
+        Assembly assembly = typeof(App).GetTypeInfo().Assembly;
+        args.ImageStream = assembly.GetManifestResourceStream("Customize_image.Assets.Image_2.png");
+    }
+    //Retrive the image from the website and use it.
```
Comment (Chrispineagunja) [inline File-Formats/DocIO/convert-markdown-to-word-document-in-csharp.md:287]: //Retrieve the image from the website and use it.

File: File-Formats/DocIO/convert-markdown-to-word-document-in-csharp.md [hunk 22]
```diff
@@ -0,0 +1,575 @@
...
+        byte[] image = client.DownloadData(args.Uri);
+        Stream stream = new MemoryStream(image);
+        //Set the retrieved image from the input Markdown.
+        args.ImageStream = stream;
+    }
+}
+{% endhighlight %}
+{% endtabs %}
+
+N> Hook the event handler before opening Word document as per above code example.
```
Comment (Chrispineagunja) [inline File-Formats/DocIO/convert-markdown-to-word-document-in-csharp.md:301]: N> Hook the event handler before opening a Word document as per the above code example.

**Pull request merged**